agenda-services/src/main/java/org/exoplatform/agenda/search/AgendaIndexingServiceConnector.java

```diff
@@ -0,0 +1,212 @@
+
+import org.exoplatform.agenda.model.Event;
+import org.exoplatform.agenda.model.EventAttendee;
+import org.exoplatform.agenda.model.Calendar;
+import org.exoplatform.agenda.service.AgendaCalendarService;
+import org.exoplatform.agenda.service.AgendaEventAttendeeService;
+import org.exoplatform.agenda.service.AgendaEventService;
+import org.exoplatform.agenda.util.AgendaDateUtils;
+import org.exoplatform.commons.search.domain.Document;
+import org.exoplatform.commons.search.index.impl.ElasticIndexingServiceConnector;
+import org.exoplatform.commons.utils.HTMLSanitizer;
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.social.core.manager.IdentityManager;
+
+public class AgendaIndexingServiceConnector extends ElasticIndexingServiceConnector {
+  private static final long           serialVersionUID = 3299213985533395198L;
+
+  public static final String          TYPE             = "event";
+
+  private static final Log            LOG              = ExoLogger.getLogger(AgendaIndexingServiceConnector.class);
+
+  private final AgendaCalendarService agendaCalendarService;
+
+  private final AgendaEventService    agendaEventService;
+
+  private AgendaEventAttendeeService  attendeeService;
+
+  public AgendaIndexingServiceConnector(AgendaCalendarService agendaCalendarService,
+                                        IdentityManager identityManager,
+                                        AgendaEventService agendaEventService,
+                                        AgendaEventAttendeeService attendeeService,
+                                        InitParams initParams) {
+    super(initParams);
+
+    this.agendaCalendarService = agendaCalendarService;
+    this.identityManager = identityManager;
+    this.agendaEventService = agendaEventService;
+    this.attendeeService = attendeeService;
+  }
+
+  @Override
+  public String getType() {
+    return TYPE;
+  }
+
+  @Override
+  public Document create(String id) {
+    return getDocument(id);
+  }
+
+  @Override
+  public Document update(String id) {
+    return getDocument(id);
+  }
+
+  @Override
+  public List<String> getAllIds(int offset, int limit) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public boolean canReindex() {
+    return false;
+  }
+
+  private Document getDocument(String id) {
+    if (StringUtils.isBlank(id)) {
+      throw new IllegalArgumentException("id is mandatory");
+    }
+    LOG.debug("Index document for agenda id={}", id);
+
+    Event event = agendaEventService.getEventById(Long.parseLong(id));
+    if (event == null) {
+      throw new IllegalStateException("event with id '" + id + "' not found");
+    }
+
+    Map<String, String> fields = new HashMap<>();
+    fields.put("id", Long.toString(event.getId()));
+
+    if (event.getParentId() > 0) {
+      fields.put("parentId", Long.toString(event.getParentId()));
```

Comment:
you can use here String.valueof so that this call can be independent from Object type (Integer, Long, short ...)